File: lib/features/pokemon/domain/usecases/search_pokemon_usecase.dart
import 'package:pokeapi/core/domain/entities/pokemon_entity.dart';
import 'package:pokeapi/core/domain/errors/pokemon_error.dart';
import 'package:pokeapi/features/pokemon/domain/repositories/pokemon_repository.dart';

/// Caso de uso para buscar Pokemons por nome
class SearchPokemonUseCase {
  final PokemonRepository _repository;

  SearchPokemonUseCase(this._repository);

  /// Executa o caso de uso
  Future<Result<List<PokemonEntity>>> call(String query) async {
    return _repository.searchPokemon(query);
  }
}

File: bff/src/domain/usecases/search-pokemon.usecase.ts
import { Pokemon } from '../entities/pokemon.entity';
import { PokemonRepository } from '../repositories/pokemon.repository';

export class SearchPokemonUseCase {
  constructor(private readonly pokemonRepository: PokemonRepository) {}

  async execute(query: string): Promise<Pokemon[]> {
    return this.pokemonRepository.searchPokemon(query);
  }
}
